src/core/control/jobs/RenderJob.cpp

```diff
@@ -79,12 +80,28 @@ void RenderJob::run() {
 
         std::lock_guard lock(this->view->drawingMutex);
         std::swap(this->view->crBuffer, newBuffer);
+        repaintPage();
```

Comment:
I guess we no longer need the lock when calling repaintPage. Maybe regroup the lock_guard and the call to std::swap in a sub-scope, so the lock gets released ASAP.